File: parallel_laboratory/Numerical_Analysis/comparing/matrix/matrix_mult.c
#include<stdio.h>
#include<stdlib.h>
#include<sys/time.h>
#include"temps1.c"
#define dim 1000
#define numar 10
int main()
{
	int i,j,k,count;
	double mata[dim][dim],matb[dim][dim],matc[dim][dim];
	struct timeval t1,t2;
	for(i=0;i<dim;i++)
	for(j=0;j<dim;j++)
	{
		mata[i][j]=rand();
		matb[i][j]=rand();
	}
	for(count=0;count<numar;count++)
	{
		gettimeofday(&t1,NULL);
		for(i=0;i<dim;i++)
		for(j=0;j<dim;j++)
		{
			matc[i][j]=0.0;
			for(k=0;k<dim;k++) matc[i][j]+=mata[i][k]*matb[k][j];
		}
		gettimeofday(&t2,NULL);
		temps1(t1,t2);
	}
}

File: parallel_laboratory/Numerical_Analysis/comparing/matrix/matrix_mult.f90
	PROGRAM matrix
	REAL(4) VAL(2)
	PARAMETER DIM=1000,NUMAR=10
	DOUBLE PRECISION MATA(DIM,DIM),MATB(DIM,DIM),MATC(DIM,DIM)
	REAL(4) TARRAY(2)
	DO I=1,DIM
	DO J=1,DIM
		MATA(I,J)=DFLOATJ(IRAND())
		MATB(I,J)=DFLOATJ(IRAND())
	END DO
	END DO
	DO L=1,NUMAR
		VAL(1)=DTIME(TARRAY)
		DO I=1,DIM
		DO J=1,DIM
			MATC(I,J)=0.0
		DO K=1,DIM
			MATC(I,J)=MATC(I,J)+MATA(I,K)*MATB(K,J)
		END DO
		END DO
		END DO
		VAL(2)=DTIME(TARRAY)
		PRINT *,TARRAY(1),'AND',TARRAY(2),'OR',VAL(2)
	END DO
	END
	